src/linear_algebra/projective_space/subspace.lean

```diff
@@ -127,6 +120,67 @@ instance : complete_lattice (subspace K V) :=
 instance subspace_inhabited : inhabited (subspace K V) :=
 { default := ⊤ }
 
+/-- The span of the empty set is the bottom of the lattice of subspaces. -/
+lemma span_empty : span (∅ : set (ℙ K V)) = ⊥ := gi.gc.l_bot
+
+/-- The span of the entire projective space is the top of the lattice of subspaces. -/
+lemma span_univ : span (set.univ : set (ℙ K V)) = ⊤ :=
+by { rw [eq_top_iff, set_like.le_def], intros x hx, exact subset_span _ (set.mem_univ x) }
+
+/-- The span of a set of points is contained in a subspace if and only if the set of points is
+contained in the subspace. -/
+lemma span_le_subspace_iff {S : set (ℙ K V)} {W : subspace K V} : span S ≤ W ↔ S ⊆ W :=
+gi.gc S W
+
+/-- If a set of points is a subset of another set of points, then its span will be contained in the
+span of that set. -/
+@[mono] lemma monotone_span : monotone (span : set (ℙ K V) → subspace K V) := gi.gc.monotone_l
+
+lemma subset_span_trans {S T U : set (ℙ K V)} (hST : S ⊆ span T) (hTU : T ⊆ span U) :
+  S ⊆ span U :=
+gi.gc.le_u_l_trans hST hTU
+
+/-- The supremum of two subspaces is equal to the span of their union. -/
+lemma span_union (S T : set (ℙ K V)) : span (S ∪ T) = span S ⊔ span T := (@gi K V _ _ _).gc.l_sup
+
+/-- The supremum of a collection of subspaces is equal to the span of the union of the
+collection. -/
+lemma span_Union {ι} (s : ι → set (ℙ K V)) : span (⋃ i, s i) = ⨆ i, span (s i) :=
+(@gi K V _ _ _).gc.l_supr
+
+/-- The supremum of a subspace and the span of a set of points is equal to the span of the union of
+the subspace and the set of points. -/
+lemma sup_span {S : set (ℙ K V)} {W : subspace K V} : W ⊔ span S = span (W ∪ S) :=
+by rw [span_union, span_coe]
+
+lemma span_sup {S : set (ℙ K V)} {W : subspace K V}: span S ⊔ W = span (S ∪ W) :=
+by rw [span_union, span_coe]
+
+/-- A point in a projective space is contained in the span of a set of points if and only if the
+point is contained in all subspaces of the projective space which contain the set of points. -/
+lemma mem_span {S : set (ℙ K V)} (u : ℙ K V) : u ∈ span S ↔ ∀ (W : subspace K V), S ⊆ W → u ∈ W :=
+by { simp_rw ← span_le_subspace_iff, exact ⟨λ hu W hW, hW hu, λ W, W (span S) (le_refl _)⟩ }
+
+/-- The span of a set of points in a projective space is equal to the infimum of the collection of
+subspaces which contain the set. -/
+lemma span_eq_Inf {S : set (ℙ K V)} : span S = Inf {W | S ⊆ W} :=
+begin
+  ext, simp_rw [mem_carrier_iff, mem_span x], refine ⟨λ hx, _, λ hx W hW, _⟩,
```

Comment:
There are exceptions but we usually keep one line per tactic:
```suggestion
  ext,
  rw [mem_carrier_iff, mem_span x],
  refine ⟨λ hx, _, λ hx W hW, _⟩,
```
See https://leanprover-community.github.io/contribute/style.html